### ISSUE TITLE
Support type narrowing with `x[K] is True/False`

### ISSUE THREAD

**Issue by trygveaa** [microsoft/pyright#4453]
**Is your feature request related to a problem? Please describe.**

If you have a discriminated union of TypedDict classes using a bool key, I have to use `x[K] == True` to narrow `x` into one of the classes. Consider the following example:

```
import json
from typing import Literal, TypedDict

class Success(TypedDict):
    ok: Literal[True]
    message: str

class Error(TypedDict):
    ok: Literal[False]
    error: str

Result = Success | Error

a: Result = json.loads('{"ok": True, "message": ""}')

if a["ok"]:
    print(a["message"]) # Error because a is still considered to be Result

if a["ok"] is True:
    print(a["message"]) # Error because a is still considered to be Result

if a["ok"] == True:
    print(a["message"]) # No error, a is correctly narrowed to Success
```

**Describe the solution you'd like**

I would like to be able to use `x[K] is True` (or False) to narrow `x` into one of the types. Or even better if I could do just `x[K]`.

**Additional context**

I see that if I have a union of normal classes rather than of TypedDicts and try to narrow with an attribute `is True` works. E.g. `if a.ok is True`.

If I have just a boolean variable I see that it's narrowed by just checking that it's truthy, e.g. after `if x`, `reveal_type(x)` reports `Literal[True]`.

I'm submitting this as a feature request rather than a bug report since I see that `x[K] is E` and `x[K]` is not listed under https://github.com/microsoft/pyright/blob/main/docs/type-concepts.md#type-guards

**Comment by erictraut** [comment]
I've added support for the forms `x[K] is V` and `x[K] is not V` (where K and V are literal expressions). I'm not adding support for the more general `x[K]` because this would impose a large performance overhead in some cases, and I don't think it's common enough to justify such a performance cost. If you want to use this pattern, you'll need to use an operator (`==`, `!=`, `is` or `is not`) in your type guard.

This change will be included in the next release.

**Comment by trygveaa** [comment]
Thanks for the quick fix!

> I'm not adding support for the more general `x[K]` because this would impose a large performance overhead in some cases, and I don't think it's common enough to justify such a performance cost. If you want to use this pattern, you'll need to use an operator (`==`, `!=`, `is` or `is not`) in your type guard.

Ah, okay.

**Comment by erictraut** [comment]
This is included in pyright 1.1.289, which I just published. It will also be included in a future release of pylance.
